Is a Dependency for ERPNext's PR #12564

saurabh-bhosale opened the pull request:
**Description:**
Added a field 'complete_address' in Address Doctype to make it searchable in Customer Link field.

![snap](https://user-images.githubusercontent.com/25682214/35174150-47772d02-fd94-11e7-881c-f5bb71d8c1b2.png)



![searchable_address](https://user-images.githubusercontent.com/25682214/35173977-9578ca0c-fd93-11e7-98ec-8d3289bebe63.gif)

saurabh-bhosale committed:
Added a field 'complete_address' in Address Doctype to make it searchable in Customer Link field

achillesrasquinha commented:
Nice. All I'm concerned is with the layout. The complete address can be hidden, or best - fetched from the Doc and formatted within the input.

saurabh-bhosale committed:
Merge branch 'develop' of https://github.com/frappe/frappe into dependency_for_address_in_online_pos

bkm888 commented:
You can see my reasoning behind needing addresses in POS search here:
https://github.com/frappe/erpnext/pull/12564

However, not every business needs this kind of function. It makes sense to be able to enable this search ability by how the customer records of a particular business model are populated. Businesses that sell products from a fixed store location would not need this so they would not populate this part of the customer record. Business that depend on route trucks to sell their products like bakeries, mechanics tools, car wash supplies, and heating fuel would use this part of the customer record to operate their mobile POS functions.

bkm@jmibarcodes.com

saurabh-bhosale committed:
Merge branch 'develop' of https://github.com/frappe/frappe into dependency_for_address_in_online_pos

saurabh-bhosale committed:
Multiple changes:
[Mod] Made the complete_address field hidden
[New] Updated patches.txt to set this field for already existing_addresses

saurabh-bhosale committed:
Codacy fix

saurabh-bhosale commented:
@achillesrasquinha Have made the field hidden in newer commits and have updated the patches too

saurabh-bhosale committed:
Merge branch 'develop' of https://github.com/frappe/frappe into dependency_for_address_in_online_pos

saurabh-bhosale committed:
[Fix] Field property issues and codacy fix

saurabh-bhosale committed:
Merge branch 'develop' of https://github.com/frappe/frappe into dependency_for_address_in_online_pos

rmehta commented on frappe/contacts/doctype/address/address.py:
```diff
@@ -59,6 +60,10 @@ def validate_reference(self):
 
+	def make_complete_address(self):
+		# for next line in link field options
+		self.complete_address = "<br>" + self.get_display()
```
why `<br>` in start?

saurabh-bhosale replied:
Used the <code>br</code> tag so that the address can be seen on newline of link field options, aimed at making it readable. Tried to see if it can be handled in the link control where this description for each item is made in html but to add it specifically before address field involved extra lines of code with different cases like an absent search_field.

1. Options with a <code>br</code>
![with_br](https://user-images.githubusercontent.com/25682214/35621336-0436e054-06aa-11e8-810d-99afa39a814d.png)
 
2. Options without a <code>br</code>
![without_br](https://user-images.githubusercontent.com/25682214/35621347-0b63ce46-06aa-11e8-94d9-48e968765ce0.png)


Also, I've changed the ERPNext's referred PR and its no more dependent on this PR, but the decision of a <code>br</code> in field versus a change in the way a link control's html is made and to introduce a <code>br</code> there is still to be made. Haven't figured out the ideal solution for it yet.

rmehta commented on frappe/contacts/doctype/address/address.json:
```diff
@@ -485,6 +485,36 @@
+   "ignore_user_permissions": 0, 
+   "ignore_xss_filter": 0, 
+   "in_filter": 0, 
+   "in_global_search": 0, 
```
you wanted it in global search?

saurabh-bhosale replied:
No, I didn't want it in global search.

saurabh-bhosale committed:
Merge branch 'develop' of https://github.com/frappe/frappe into dependency_for_address_in_online_pos
Resolved Conflicts:frappe/patches.txt

saurabh-bhosale committed:
[Mod] Updated address.json

rohitwaghchaure commented on frappe/patches/v10_0/make_complete_address.py:
```diff
@@ -0,0 +1,10 @@
+		address_doc = frappe.get_doc("Address",address)
+		address_doc.make_complete_address()
+		address_doc.save()
+		frappe.db.commit()
```
No need to commit here

saurabh-bhosale replied:
@rohitwaghchaure Have made the suggested changes in newer commits.

saurabh-bhosale committed:
Merge branch 'develop' of https://github.com/frappe/frappe into dependency_for_address_in_online_pos

saurabh-bhosale committed:
[Mod] Edited address.py and removed commit from patch file 'make_complete_address.py'

rohitwaghchaure commented on frappe/patches/v10_0/make_complete_address.py:
```diff
@@ -0,0 +1,9 @@
+	for address in existing_addresses:
+		address_doc = frappe.get_doc("Address",address)
+		address_doc.make_complete_address()
+		address_doc.save()
```
Don't save address form, just update the field complete_address. It will save the time during execution of the patch

saurabh-bhosale committed:
Merge branch 'develop' of https://github.com/frappe/frappe into dependency_for_address_in_online_pos

saurabh-bhosale committed:
[Mod] Updated 'make_complete_address' patch

saurabh-bhosale committed:
Used db_set method instead of save in patch 'make_complete_address.py'

saurabh-bhosale committed:
[Fix] Code correction

saurabh-bhosale commented:
Removed the dependency on this PR. Therefore, closing it.
